src/partial-result-stream.ts

```diff
@@ -50,11 +50,30 @@ interface RequestFunction {
  *     that it is not ready for any more data. Increase this value if you
  *     experience 'Stream is still not ready to receive data' errors as a
  *     result of a slow writer in your receiving stream.
+ * @property {object} [columnsMetadata] An object map that can be used to pass
+ * additional properties for each column type which can help in deserializing
+ * the data coming from backend. (Eg: We need to pass Proto Function and Enum
+ * map to deserialize proto messages and enums, respectively.)
```

Comment:
This and also the comment below on `columnsMetadata` is too abstract. Could we add a concrete example of what we expect here?